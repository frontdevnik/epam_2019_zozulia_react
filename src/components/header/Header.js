import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { Routes } from '../../constans';
import withTranslation from '../../hocs/withTranslation';
import styles from './header.module.scss';

function Header(props) {
  let homeLink = <div />;
  let loginLink;
  const { isLoggin, header_home_link, header_logout_link, header_login_link, header_registration_link, header_title } = props;

  const logOut = () => {
    localStorage.setItem('isLoggin', 0);
    window.location.reload(true);
  };

  const changeLang = (lang) => () => {
    localStorage.setItem('selectedLanguage', lang);
    window.location.reload(true);
  }

  if (isLoggin) {
    homeLink = (
      <div>
        <NavLink className={styles.link} to={Routes.HOMEPAGE}>{header_home_link}</NavLink>
      </div>
    );
    loginLink = (
      <p onClick={logOut} className={styles.link}>{header_logout_link}</p>
    );
  } else {
    loginLink = (
      <div>
        <NavLink className={styles.link} activeClassName={styles.active} to={Routes.REGISTRATION}>{header_registration_link}</NavLink>
        <NavLink className={styles.link} activeClassName={styles.active} to={Routes.LOGIN}>{header_login_link}</NavLink>
      </div>
    );
  }
  return (
    <>
      <header className={styles.header}>
        <div className={styles.langContainer}>
          <span onClick={changeLang('ukr')} className={styles.lang}>ukr </span>/
          <span onClick={changeLang('eng')} className={styles.lang}> eng</span>
        </div>
        {homeLink}
        {loginLink}
      </header>
      {isLoggin ? <h1 className={styles.title}>{header_title}</h1> : null}
    </>
  );
}

Header.propTypes = {
  isLoggin: PropTypes.bool,
};

const mapStateToProps = (state) => ({
  isLoggin: state.loginReducer.isLoggin,
});

const withConnect = connect(mapStateToProps);
const withTranslationWords = withTranslation(['header_home_link', 'header_logout_link', 'header_login_link', 'header_registration_link', 'header_title']);

export default withTranslationWords(withConnect(Header));
