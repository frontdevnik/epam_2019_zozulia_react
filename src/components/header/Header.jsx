import React from 'react';
import { NavLink } from 'react-router-dom';
import { useSelector } from 'react-redux';
import withTranslation from '../../hocs/withTranslation';

import {
  HOME_PAGE,
  LOGIN_PAGE,
  REGISTRATION_PAGE,
} from '../../constants/path-constans';

import styles from './header.module.scss';

const Header = ({
  header_home_link,
  header_logout_link,
  header_login_link,
  header_registration_link,
  header_title,
}) => {
  const isLoggin = useSelector((state) => state.loginReducer.isLoggin);

  const logOut = () => {
    localStorage.setItem('isLoggin', 0);
    window.location.reload(true);
  };

  const changeLang = (lang) => () => {
    localStorage.setItem('selectedLanguage', lang);
    window.location.reload(true);
  };

  return (
    <>
      <header className={styles.header}>
        <div className={styles.langContainer}>
          <span onClick={changeLang('ukr')} className={styles.lang}>ukr </span>
          /
          <span onClick={changeLang('eng')} className={styles.lang}> eng</span>
        </div>
        {isLoggin ? <NavLink className={styles.link} to={HOME_PAGE}>{header_home_link}</NavLink> : <div />}
        {isLoggin
          ? <p onClick={logOut} className={styles.link}>{header_logout_link}</p>
          : (
            <div>
              <NavLink
                className={styles.link}
                activeClassName={styles.active}
                to={REGISTRATION_PAGE}
              >
                {header_registration_link}
              </NavLink>
              <NavLink
                className={styles.link}
                activeClassName={styles.active}
                to={LOGIN_PAGE}
              >
                {header_login_link}
              </NavLink>
            </div>
          )}
      </header>
      {isLoggin ? <h1 className={styles.title}>{header_title}</h1> : null}
    </>
  );
};

const withTranslationWords = withTranslation(['header_home_link', 'header_logout_link', 'header_login_link', 'header_registration_link', 'header_title']);

export default withTranslationWords(Header);
