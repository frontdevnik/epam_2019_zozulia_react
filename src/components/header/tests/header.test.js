import React from 'react';
import {useSelector} from "react-redux";
import {
    BrowserRouter
} from "react-router-dom";
import {TestHeader} from "../Header";
import {headerProps} from "./stubProps";
import styles from '../header.module.scss';


const setUp = props => mount(<BrowserRouter><TestHeader {...props} /></BrowserRouter>);

jest.mock('react-redux', () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
}));

describe('<Header />', () => {
    describe('should render correctly when user is not login', () => {
        let component;

        beforeEach(() => {
            useSelector.mockReturnValue(false);
            component = setUp(headerProps);
        });

        it('should render', () => {
            expect(component.html()).toMatchSnapshot();
        });

        it('should render registration link', () => {
            expect(component.find('a').first().text()).toBe(headerProps.header_registration_link);
        });

        it('should render login link', () => {
            expect(component.find('a').last().text()).toBe(headerProps.header_login_link);
        });
    });

    describe('should render correctly when user login', () => {
        let component;

        beforeEach(() => {
            useSelector.mockReturnValue(true);
            component = setUp(headerProps);
        });

        it('should render', () => {
            expect(component.html()).toMatchSnapshot();
        });

        it('should render home link', () => {
            expect(component.find('a').text()).toBe(headerProps.header_home_link);
        });

        it('should render logout link', () => {
            expect(component.find('p').text()).toBe(headerProps.header_logout_link);
        });

        it('should render title', () => {
            expect(component.find('h1').text()).toBe(headerProps.header_title);
        });
    })
})
