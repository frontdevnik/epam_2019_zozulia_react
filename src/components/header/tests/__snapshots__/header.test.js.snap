// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `"<header class=\\"header\\"><div class=\\"langContainer\\"><span class=\\"lang\\">ukr </span>/<span class=\\"lang\\"> eng</span></div><a class=\\"link\\" href=\\"/home\\">home</a><p class=\\"link\\">logout</p></header><h1 class=\\"title\\">title</h1>"`;

exports[`<Header /> should render correctly when user is not login render 1`] = `"<header class=\\"header\\"><div class=\\"langContainer\\"><span class=\\"lang\\">ukr </span>/<span class=\\"lang\\"> eng</span></div><div></div><div><a class=\\"link\\" href=\\"/registration\\">registration</a><a class=\\"link\\" href=\\"/login\\">login</a></div></header>"`;

exports[`<Header /> should render correctly when user is not login should render 1`] = `"<header class=\\"header\\"><div class=\\"langContainer\\"><span class=\\"lang\\">ukr </span>/<span class=\\"lang\\"> eng</span></div><div></div><div><a class=\\"link\\" href=\\"/registration\\">registration</a><a class=\\"link\\" href=\\"/login\\">login</a></div></header>"`;

exports[`<Header /> should render correctly when user login render 1`] = `"<header class=\\"header\\"><div class=\\"langContainer\\"><span class=\\"lang\\">ukr </span>/<span class=\\"lang\\"> eng</span></div><a class=\\"link\\" href=\\"/home\\">home</a><p class=\\"link\\">logout</p></header><h1 class=\\"title\\">title</h1>"`;

exports[`<Header /> should render correctly when user login should render 1`] = `"<header class=\\"header\\"><div class=\\"langContainer\\"><span class=\\"lang\\">ukr </span>/<span class=\\"lang\\"> eng</span></div><a class=\\"link\\" href=\\"/home\\">home</a><p class=\\"link\\">logout</p></header><h1 class=\\"title\\">title</h1>"`;

exports[`<Header /> should render correctly2 1`] = `"<header class=\\"header\\"><div class=\\"langContainer\\"><span class=\\"lang\\">ukr </span>/<span class=\\"lang\\"> eng</span></div><div></div><div><a class=\\"link\\" href=\\"/registration\\">registration</a><a class=\\"link\\" href=\\"/login\\">login</a></div></header>"`;
